xcdat/averager_accessor.py

```diff
@@ -0,0 +1,505 @@
+        )
+        bounds: Dict[str, BoundsByType] = {
+            "lat": {
+                "domain": self._dataset.bounds.get_bounds("lat"),
+                "region": lat_bounds,
+            },
+            "lon": {
+                "domain": self._dataset.bounds.get_bounds("lon"),
+                "region": lon_bounds,
+            },
+        }
+        weights: AxisWeights = {}
+
+        for dim, bounds_by_type in bounds.items():
+            domain_bounds = bounds_by_type["domain"]
+            region_bounds = bounds_by_type["region"]
+
+            if domain_bounds is not None:
+                domain_bounds_copy = domain_bounds.copy()
+
+                if region_bounds is not None:
+                    # ensure region bounds can be indexed and are type(float)
+                    region_bounds_arr = np.array(region_bounds).astype(float)
+
+                    # TODO: Refactor this chunk of code into a private function
+                    # =========================================================
+                    if dim == "lon":
+                        # swap region + dim bounds to be 360-type
+                        domain_bounds_copy[:] = self._swap_region_lon_axis(
+                            domain_bounds, "360"
+                        )
+                        region_bounds_arr[:] = self._swap_region_lon_axis(
+                            region_bounds_arr, "360"
+                        )
+                        # find any instances of an upper lon bound being less than a
+                        # lower bound, e.g., across prime meridian or dateline,
+                        # [0.625, 359.375]
+                        rowIndex = np.where(
+                            (domain_bounds_copy[:, 1] - domain_bounds_copy[:, 0]) < 0
+                        )[0]
+                        # if applicable, force this indexed row to be linearly
+                        # continuous
+                        if len(rowIndex) > 0:
+                            domain_bounds_copy[rowIndex, 1] = (
+                                domain_bounds_copy[rowIndex, 1] + 360.0
+                            )
+                        # do the same for region bounds
+                        regionInds = np.where(
+                            region_bounds_arr < float(np.min(domain_bounds_copy))
+                        )[0]
+                        if len(regionInds) > 0:
+                            region_bounds_arr[regionInds] = (
+                                region_bounds_arr[regionInds] + 360.0
+                            )
+                        # if the region bounds are equal, it means all longitudes are
+                        # specified (e.g., [0, 360] -> [360, 360]). In this case, make
+                        # the region bounds the upper/lower values of the dim bounds
+                        if region_bounds_arr[1] - region_bounds_arr[0] == 0:
+                            region_bounds_arr[1] = np.max(domain_bounds)
```

Comment:
@tomvothecoder - I think this one needs to be updated to `domain_bounds_copy` too. I think this should work after that update.

Reply:
Thanks for the catch again, darn VSCode renaming lol.